packages/react-divider/src/components/Divider/renderDivider.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { getSlots, useId } from '@fluentui/react-utilities';
+import { DividerState } from './Divider.types';
+import { dividerShorthandProps } from './useDivider';
+
+/**
+ * Function that renders the final JSX of the component
+ */
+export const renderDivider = (state: DividerState) => {
+  const { slots, slotProps } = getSlots(state, dividerShorthandProps);
+  const { children } = state;
+  const idBase = children ? useId('divider-') : undefined;
```

Comment:
this should be stored in the state hook. Render functions in principle are supposed to be as simple as possible.

If a consumer were to create a variant of the divider component they should not have to take a dependency on `useId`

Reply:
Updated and moved to the state.